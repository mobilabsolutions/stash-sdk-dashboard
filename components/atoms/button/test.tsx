/* eslint-env jest */

import { testRender } from '../../../test_utils'
import PrimaryButton from './primary'
import SecondaryButton from './secondary'
import WarnButton from './warn'

it('PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Login' }))

it('Disabled PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Disabled', disabled: true }))

it('Full size PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Disabled', isFullSize: true }))

it('SecondaryButton Should Render', () =>
  testRender(SecondaryButton, { lable: 'Cancel' }))

it('Disabled PrimaryButton Should Render', () =>
  testRender(SecondaryButton, { lable: 'Cancel', disabled: true }))

it('WarnButton Should Render', () =>
  testRender(WarnButton, { lable: 'Cancel' }))

it('Disabled WarnButton Should Render', () =>
  testRender(WarnButton, { lable: 'Cancel', disabled: true }))
