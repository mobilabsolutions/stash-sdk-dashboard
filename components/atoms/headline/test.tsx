/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { withTestSetup } from '../../../test_utils'
import H1 from './h1'
import H2 from './h2'

const TestH1 = withTestSetup(H1)
const TestH2 = withTestSetup(H2)

it('H1 Should Render', () => {
  const component = renderer.create(<TestH1>Heading 1</TestH1>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('H2 Should Render', () => {
  const component = renderer.create(<TestH2>Heading 2</TestH2>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
