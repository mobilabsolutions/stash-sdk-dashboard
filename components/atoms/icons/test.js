/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { withTestSetup } from '../../../test_utils'
import DownIcon from './down'
import UpIcon from './up'

const TestDownIcon = withTestSetup(DownIcon)
const TestUpIcon = withTestSetup(UpIcon)

it('Down Icon Should Render', () => {
  const component = renderer.create(<TestDownIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('UpIcon Should Render', () => {
  const component = renderer.create(<TestUpIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
