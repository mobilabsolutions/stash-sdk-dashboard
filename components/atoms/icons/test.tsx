/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { withTestSetup } from '../../../test_utils'
import Logo from './logo'
import AccountIcon from './account'
import KeyIcon from './key'
import CopyIcon from './copy'
import DownIcon from './down'
import UpIcon from './up'

const TestLogo = withTestSetup(Logo)
const TestAccountIcon = withTestSetup(AccountIcon)
const TestKeyIcon = withTestSetup(KeyIcon)
const TestCopyIcon = withTestSetup(CopyIcon)
const TestDownIcon = withTestSetup(DownIcon)
const TestUpIcon = withTestSetup(UpIcon)

it('Logo Should Render', () => {
  const component = renderer.create(<TestLogo />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('AccountIcon Should Render', () => {
  const component = renderer.create(<TestAccountIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('KeyIcon Should Render', () => {
  const component = renderer.create(<TestKeyIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('CopyIcon Should Render', () => {
  const component = renderer.create(<TestCopyIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('UpIconIcon Should Render', () => {
  const component = renderer.create(<TestUpIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('DownIcon Should Render', () => {
  const component = renderer.create(<TestDownIcon />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
