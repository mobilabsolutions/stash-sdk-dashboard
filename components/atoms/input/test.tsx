import { testRender } from '../../../test_utils'

import Input from './input'
import VisibilityIcon from './visibility_icon'

it('Input Should Render', () => testRender(Input))
it('VisibilityIcon Should Render', () => testRender(VisibilityIcon))
