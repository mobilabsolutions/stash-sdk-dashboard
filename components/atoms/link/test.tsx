/* eslint-env jest */

import { testRender } from '../../../test_utils'
import Link from './index'

it('Link Should Render', () =>
  testRender(Link, { lable: 'Login', href: 'https://google.com' }))
