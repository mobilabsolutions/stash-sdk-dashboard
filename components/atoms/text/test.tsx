/* eslint-env jest */

import { testRender } from '../../../test_utils'

import Title from './title'

import H1 from './h1'
import H2 from './h2'
import H3 from './h3'
import H4 from './h4'

import ActiveH1 from './h1_active'
import ActiveH2 from './h2_active'
import ActiveH3 from './h3_active'
import ActiveH4 from './h4_active'

import Body from './body'
import LargeBody from './body_large'
import PlusBody from './body_plus'
import SmallBody from './body_small'

import Warning from './warning'

it('Title Should render', () => testRender(Title))

it('H1 Should render', () => testRender(H1))
it('H2 Should render', () => testRender(H2))
it('H3 Should render', () => testRender(H3))
it('H4 Should render', () => testRender(H4))

it('ActiveH1 Should render', () => testRender(ActiveH1))
it('ActiveH2 Should render', () => testRender(ActiveH2))
it('ActiveH3 Should render', () => testRender(ActiveH3))
it('ActiveH4 Should render', () => testRender(ActiveH4))

it('Body Should render', () => testRender(Body))
it('LargeBody Should render', () => testRender(LargeBody))
it('PlusBody Should render', () => testRender(PlusBody))
it('SmallBody Should render', () => testRender(SmallBody))

it('Warning Should render', () => testRender(Warning))
