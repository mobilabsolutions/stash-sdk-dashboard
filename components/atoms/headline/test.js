/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import { withTestSetup } from '../../test'
import H1 from './h1'
import H2 from './h2'
import H3 from './h3'
import H4 from './h4'
import H5 from './h5'

const TestH1 = withTestSetup(H1)
const TestH2 = withTestSetup(H2)
const TestH3 = withTestSetup(H3)
const TestH4 = withTestSetup(H4)
const TestH5 = withTestSetup(H5)

it('H1 Should Render', () => {
  const component = renderer.create(<TestH1>Heading 1</TestH1>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('H2 Should Render', () => {
  const component = renderer.create(<TestH2>Heading 2</TestH2>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('H3 Should Render', () => {
  const component = renderer.create(<TestH3>Heading 3</TestH3>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('H4 Should Render', () => {
  const component = renderer.create(<TestH4>Heading 4</TestH4>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('H5 Should Render', () => {
  const component = renderer.create(<TestH5>Heading 5</TestH5>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
