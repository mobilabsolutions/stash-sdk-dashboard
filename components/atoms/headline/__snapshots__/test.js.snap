// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Should Render 1`] = `
<h1
  className="h1-sc-1dhgt1i-0 ktdQQC"
>
  Heading 1
</h1>
`;

exports[`H2 Should Render 1`] = `
<h2
  className="h2-sc-1d7ooui-0 iIfZtO"
>
  Heading 2
</h2>
`;

exports[`H3 Should Render 1`] = `
<h3
  className="h3-h296jt-0 idFdrZ"
>
  Heading 3
</h3>
`;

exports[`H4 Should Render 1`] = `
<h4
  className="h4-zwc3qv-0 qVbje"
>
  Heading 4
</h4>
`;

exports[`H5 Should Render 1`] = `
<h5
  className="h5-ylrd96-0 cbVLSS"
>
  Heading 5
</h5>
`;
